### PR TITLE
6 get article comments by article

### PR DESCRIPTION
**Pull request by DexterGundar**
I know it has got task 5 commits under it, my merge did not work correctly for task 5, next time before merging I will nchelp.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -116,3 +115,58 @@ describe('get articles',()=>{
...
+            body.comments.forEach((comment) => {
+              expect(comment).toHaveProperty("comment_id", expect.any(Number)),
+              expect(comment).toHaveProperty("body", expect.any(String)),
+              expect(comment).toHaveProperty("votes", expect.any(Number)),
+              expect(comment).toHaveProperty("author", expect.any(String)),
+              expect(comment).toHaveProperty("created_at", expect.any(String)),
+              expect(comment).toHaveProperty("article_id", expect.any(Number))
+            })
+      })
+  });
```
Comment (elijaharley) [inline __tests__/app.test.js:135]: We can be explicit here about what the article id is as we want to make sure we only have comments from this article.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -116,3 +115,58 @@ describe('get articles',()=>{
...
+  })
+  test("return 404 and message that an article does not exist if article ID is not in db", () => {
+    return request(app)
+        .get("/api/articles/5555/comments")
+        .expect(404)          
+        .then(({ body }) => {
+        expect(body.message).toBe('This article does not exist')
+        })
+  });
+})
```
Comment (elijaharley) [inline __tests__/app.test.js:172]: Some good testing here, I'd suggest keeping all 200 tests together and error handling tests below just to make it more readable.

File: controllers/app.controllers.js [hunk 3]
```diff
@@ -13,6 +13,18 @@ exports.getTopics = (req, res, next) => {
     .catch(next)
 }
 
+exports.getArticleById = (req, res, next) => {
+    const {article_id} = req.params
+    if (isNaN(article_id)) return next({ status: 400, message: 'Not a number, please enter valid id'});
+
```
Comment (elijaharley) [inline controllers/app.controllers.js:19]: This logic will be handled by psql and can be caught by your psql error handlers.

File: controllers/app.controllers.js [hunk 4]
```diff
@@ -23,15 +35,14 @@ exports.getArticles = (req, res, next) =>{
     })
 }
 
-exports.getArticleById = (req, res, next) => {
+exports.getCommentsByArtId = (req, res, next) =>{
     const {article_id} = req.params
     if (isNaN(article_id)) return next({ status: 400, message: 'Not a number, please enter valid id'});
```
Comment (elijaharley) [inline controllers/app.controllers.js:40]: See above.

File: models/app.models.js [hunk 5]
```diff
@@ -37,4 +37,43 @@ exports.fetchArticles = () =>{
...
+        return [
+          {
+            message: "There are no comments associated with this article"
+          },
+        ];
+      } else {
+        return rows;
+      }
+    });
+});
```
Comment (elijaharley) [inline models/app.models.js:78]: I like what you're thinking here with checking the article exists. We already have a model that gets the article by its id so we can handle this logic using a Promise.all in the controller to check the article exists and update it at the same time.

**Review by elijaharley — CHANGES_REQUESTED** [review]
Some nice work on this ticket. Please see my comments inline and refactor accordingly.

File: __tests__/app.test.js [hunk 6]
```diff
@@ -116,3 +115,58 @@ describe('get articles',()=>{
...
+    });
+  });
+  test("return 200 and message that there are no comments associated with this article", () => {
+    return request(app)
+    .get("/api/articles/4/comments")
+    .expect(200)          
+    .then(({ body }) => {
+      expect(body.comments).toEqual([
+        {
+          message: 'There are no comments associated with this article'
```
Comment (B7Bprog) [inline __tests__/app.test.js:151]: My suggestion here is to expect an empty array as it's a happy path. I leave it up to you to refactor it or not.

File: controllers/app.controllers.js (hunk 3 above)
Comment (B7Bprog) [inline controllers/app.controllers.js:19]: Just as Eli said, you could rely on Psql for this one.

File: controllers/app.controllers.js [hunk 7]
```diff
@@ -23,15 +35,16 @@ exports.getArticles = (req, res, next) =>{
...
-
-    fetchArticleById(article_id)
-    .then((article)=>{
-        res.status(200).send({ article })
+    fetchArticleById(article_id).then(()=>{
+        fetchCommentsByArtId(article_id)
```
Comment (B7Bprog) [inline controllers/app.controllers.js:42]: Please eliminate the nesting of .then()'s here.

File: models/app.models.js [hunk 8]
```diff
@@ -37,4 +37,43 @@ exports.fetchArticles = () =>{
...
         return rows
     })
+}
+
+exports.fetchCommentsByArtId = (article_id)=>{
+    const articleString = `
+    SELECT * FROM articles
+    WHERE article_id = $1;
+`;
+return db
```
Comment (B7Bprog) [inline models/app.models.js:47]: You already trying to get an article with that ID from your controller using the original function for that. Is it necessary to get that article again here?

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Gundars, please consider out suggestions in our comments, then it's good to be merged.

File: __tests__/app.test.js (hunk 6 above)
Comment (DexterGundar) [reply __tests__/app.test.js:151]: done

File: models/app.models.js (hunk 8 above)
Comment (DexterGundar) [reply models/app.models.js:47]: done.

File: controllers/app.controllers.js (hunk 3 above)
Comment (DexterGundar) [reply controllers/app.controllers.js:19]: done

File: controllers/app.controllers.js (hunk 7 above)
Comment (DexterGundar) [reply controllers/app.controllers.js]: done

**Review by B7Bprog — APPROVED** [review]
Hey Gundars, thanks for the updates, please merge.

**Pull request merged**